### ISSUE TITLE
Different order in files passed from the command line produces different (and incorrect) output

### ISSUE THREAD

**Issue by andreaTP** [PyCQA/pylint#3661]
### Steps to reproduce

I prepared a minimal repro here:
https://github.com/andreaTP/pylint_order_bug

### Current behavior

Cloning the example project and running:
```bash
python3 -m pylint test2/test3/test.py test2/test4/test.py test/__init__.py
```
R0801 (similarity) is reported on an incorrect file:
```
test/__init__.py:1:0: R0801: Similar lines in 2 files
```

### Expected behavior

Cloning the example project and running:
```bash
python3 -m pylint test2/test3/test.py test2/test4/test.py test/__init__.py
```
 produces the same result as:
```bash
python3 -m pylint test/__init__.py test2/test3/test.py test2/test4/test.py
```
Such as `R0801` is reported on the correct file:
```
test2/test4/test.py:1:0: R0801: Similar lines in 2 files
```

### pylint --version output
```
pylint 2.5.2
astroid 2.4.1
Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0]
```

**Comment by Pierre-Sassoulas** [comment]
Seems somewhat similar to #4053

**Comment by Pierre-Sassoulas** [comment]
I'm going to close as a duplicate of #4053, the fix will probably be the same.
